x-pack/platform/plugins/shared/onechat/public/application/components/tools/esql/esql_tool.tsx

```diff
@@ -107,6 +116,45 @@ export const EsqlTool: React.FC<EsqlToolProps> = ({
     </EuiButton>
   );
 
+  const saveAndTestButton = (
+    <EuiButton
+      fill
+      onClick={async () => {
+        const formData = form.getValues();
+        let savedTool;
+        if (mode === OnechatEsqlToolFormMode.Edit) {
+          savedTool = await saveTool(transformEsqlFormDataForUpdate(formData));
+        } else {
+          savedTool = await saveTool(transformEsqlFormDataForCreate(formData));
+        }
+        if (savedTool?.id) {
+          setCurrentToolId(savedTool.id);
```

Comment:
can't we just use `watch` from react hook form  and have `const toolId = watch("toolId") ` (or whatever reference is in form)this avoids using extra state for id storing.

 All you need to do is save tool and show flyout and this should just work then